### ISSUE TITLE
JsonEncoder should allow sequences to special-case null behaviour, like when writing objects

### ISSUE THREAD

**Issue by regiskuckaertz** [zio/zio-json#70]
i.e. `encode(None) == ''` instead of `encode(None) == null`? Doable with small changes to the object and array encoders. Ideally this would be a configurable behaviour. What do you think?

**Comment by fsvehla** [comment]
Are you thinking about `Option[String]` emitting an empty string or all `None`?

**Comment by regiskuckaertz** [comment]
I mean `Option[A]` yields the empty string or the encoding of `A`, yes. So `List(Some(1), Some(2), None, Some(4))` is turned into `[1, 2, 4]` and not `[1, 2, null, 4]`. We have an example at work where a client-side library is using this huge model for data tables, and 90% of the properties are going to be `null` most of the time: not serialising them cuts down the payload by a lot.

I know that would break the roundtrip property but I feel this rule should be expressed as `decode(encode(xs)) == normalise(xs)`.

**Comment by fommil** [comment]
I don't think it would be wise to make this the default since it will change the contents and length of the sequence in the general case, and position might be important.

If I was faced with the problem, I would use the `.contramap` mechanism to `.flatten` the `Seq[Maybe[A]]` to a `Seq[A]`.

If a more general solution is needed (which would also be more efficient), I would create variants of the sequence encoders and make them `JsonEncoder.isNothing` aware of the contents.

So I think this is best resolved by updating the README to include this as a (really useful) motivating example of the power of the `.contramap` derivation mechanism, and/or providing locally scoped implicits to override the sequence encoder defaults.

**Comment by regiskuckaertz** [comment]
Good point on the position. I took the example of a sequence but it would also apply to objects where null values are simply dropped from the object. We use a table data model that is quite verbose at work and doing that results in massive gains.

Definitely against making this a default behaviour, though when I looked at the code I thought it could go from"

```scala
def unsafeEncode(a: A, indent: Option[Int], out: Write): Unit
```
to
```
case class EncoderSettings(
  indent: Option[Iint],
  ignoreNull: Boolean
)
def unsafeEncode(a: A, settings: EncoderSettings, out: Write): Unit
```
